packages/pcd/pod-ticket-pcd/test/pod-ticket-pcd.spec.ts

```diff
@@ -0,0 +1,129 @@
+      },
+      privateKey: {
+        value: prvKey,
+        argumentType: ArgumentTypeName.String
+      },
+      id: {
+        value: undefined,
+        argumentType: ArgumentTypeName.String
+      }
+    });
+  });
+
+  it("should be able to create and verify a signed ticket", async function () {
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.true;
+  });
+
+  it("should not be possible to verify a ticket that has been tampered with", async function () {
+    const originalTicketData = ticket.claim.ticket;
+    ticket.claim.ticket = {
+      ...originalTicketData,
+      attendeeEmail: "hacker@example.com"
+    };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = {
+      ...originalTicketData,
+      attendeeName: "Not The Ticket Holder"
+    };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = { ...originalTicketData, eventId: uuid() };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = { ...originalTicketData, productId: uuid() };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = { ...originalTicketData, ticketId: uuid() };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = {
+      ...originalTicketData,
+      attendeeSemaphoreId: "54321"
+    };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = { ...originalTicketData, isConsumed: true };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = { ...originalTicketData, isRevoked: true };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = {
+      ...originalTicketData,
+      timestampConsumed: 0
+    };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = {
+      ...originalTicketData,
+      timestampSigned: 0
+    };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    ticket.claim.ticket = {
+      ...originalTicketData,
+      ticketCategory: TicketCategory.PcdWorkingGroup
+    };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.false;
+
+    // Just to show that the original data definitely still works
+    ticket.claim.ticket = { ...originalTicketData };
+    expect(await PODTicketPCDPackage.verify(ticket)).to.be.true;
+  });
+
+  it("should be possible to serialize and deserialize the pcd", async function () {
+    const serialized = await PODTicketPCDPackage.serialize(ticket);
+    const deserialized = await PODTicketPCDPackage.deserialize(serialized.pcd);
+    expect(deserialized.claim).to.deep.eq(ticket.claim);
+    expect(deserialized.proof).to.deep.eq(deserialized.proof);
+    expect(deserialized.type).to.eq(deserialized.type);
+    expect(deserialized.id).to.eq(deserialized.id);
+  });
```

Comment:
Should maybe confirm that the deserialized PCD can be verified too?  Though if the proof and claim are identical, that's implied if there weren't something weird in the implementation.